From c7a277e6810b74b7f664b029573beab592932421 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Mon, 1 Aug 2022 22:10:02 +0200
Subject: [PATCH] Drop obsolete X11 dependency, introduce WITH_X11 option
 instead

This will only determine whether to search for Qt5X11Extras and then
set HAVE_X11 accordingly. This is a behavior change as previously it was
silently disabled if X11 was not found.

We want to be able to build without X11 support even if some of the used
libraries may not work w/o X11 themselves yet or need to be built with
X11 support for other reverse dependencies.

HAVE_X11 already exists and is set automagically so far, but using
-DCMAKE_DISABLE_FIND_PACKAGE_X11 will break if any dependencies list X11
as required in their cmake config.

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 CMakeLists.txt                   | 19 +++++++++----------
 src/ioslaves/http/CMakeLists.txt |  3 ---
 2 files changed, 9 insertions(+), 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 95b5db299..c3372f7ea 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -117,17 +117,16 @@ set_package_properties(GSSAPI PROPERTIES DESCRIPTION "Allows KIO to make use of
                        PURPOSE "A MIT or HEIMDAL flavor of GSSAPI can be used"
                       )
 
-if (NOT APPLE AND NOT WIN32)
-    find_package(X11)
-endif()
-
-set(HAVE_X11 ${X11_FOUND})
-if (HAVE_X11)
-    if (QT_MAJOR_VERSION STREQUAL "5")
-        find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED X11Extras)
-    else()
-        # qtx11extras_p.h is in Qt6Gui, which is implied by the find_package(QtWidgets) above
+if (NOT APPLE AND NOT WIN32 AND NOT ANDROID)
+    option(WITH_X11 "Build with X11 integration" ON)
+    if(WITH_X11)
+        if (QT_MAJOR_VERSION STREQUAL "5")
+            find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED X11Extras)
+        else()
+            # qtx11extras_p.h is in Qt6Gui, which is implied by the find_package(QtWidgets) above
+        endif()
     endif()
+    set(HAVE_X11 ${WITH_X11})
 endif()
 
 find_package(ACL)
diff --git a/src/ioslaves/http/CMakeLists.txt b/src/ioslaves/http/CMakeLists.txt
index 4fbddb4a3..689f63dd8 100644
--- a/src/ioslaves/http/CMakeLists.txt
+++ b/src/ioslaves/http/CMakeLists.txt
@@ -3,9 +3,6 @@ include(ECMMarkNonGuiExecutable)
 include(ConfigureChecks.cmake)
 configure_file(config-kioslave-http.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-kioslave-http.h )
 
-find_package(X11)
-set(HAVE_X11 ${X11_FOUND})
-
 if(GSSAPI_FOUND)
     set(HAVE_LIBGSSAPI 1)
     if(GSSAPI_FLAVOR STREQUAL "MIT")
-- 
GitLab

